src/wasm.h

```diff
@@ -1427,15 +1427,15 @@ class Function {
   }
 };
 
+enum class Kind {
```

Comment:
Hmm, I prefer `Kind` I think. I'm not sure we need this just for external things, e.g. a function neither imported nor exported still has a "Function" kind, and some passes might need that (e.g. I'm planning a globalopt pass which be in that situation). And the design repo uses both `external_kind` and `kind`, but mostly the latter. Finally, `Kind` is nice and short, and I don't think at risk of colliding with anything, at least that I can think of.